src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -75,15 +75,44 @@ file(
   frontend/src/*/*/*/*/*/*.ts
   frontend/src/*/*/*/*/*/*.html)
 
+# these files are generated during build
+list(REMOVE_ITEM frontend_src
+  ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend/src/environments/environment.prod.ts
+  ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend/src/environments/environment.ts)
+
+execute_process(
+    COMMAND bash -c "cat ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend/package.json | jq -r .config.locale"
+    OUTPUT_VARIABLE default_lang)
+string(STRIP "${default_lang}" default_lang)
+
+if (DASHBOARD_FRONTEND_LANGS)
+  string(STRIP "${DASHBOARD_FRONTEND_LANGS}" DASHBOARD_FRONTEND_LANGS)
+  if("${DASHBOARD_FRONTEND_LANGS}" MATCHES "ALL")
+    set(build_target ":*")
+  else()
+    string(REPLACE " " "," build_target "${DASHBOARD_FRONTEND_LANGS}")
+    string(FIND "${build_target}" "${default_lang}" default_idx)
+    if (default_idx EQUAL -1)
+      # default language must be always built
+      string(CONCAT build_target "${build_target}" ",${default_lang}")
+    endif()
+    set(build_target ":\{${build_target}\}")
+  endif()
+else(DASHBOARD_FRONTEND_LANGS)
+  set(build_target ":${default_lang}")
+endif(DASHBOARD_FRONTEND_LANGS)
+
+message(STATUS "Will build dashboard frontend with the following script: build${build_target}")
```

Comment:
i'd suggest add this to the COMMENT of the npm command which builds the frontmend.